src/apps/basic/basic_apps.lua

```diff
@@ -6,11 +6,40 @@ local app = require("core.app")
 local freelist = require("core.freelist")
 local packet = require("core.packet")
 local link = require("core.link")
+local ffi = require("ffi")
 local transmit, receive = link.transmit, link.receive
 
+--- # `Truncate` app: truncate or zero pad packet to length n
+Truncate = {}
+function Truncate:new(n)
+   return setmetatable({n = n}, {__index=Truncate})
+end
+function Truncate:push()
+   for _ = 1, link.nreadable(self.input.input) do
+      local p = receive(self.input.input)
+      ffi.fill(p.data, math.min(0, self.n - p.length))
```

Comment:
http://snabbco.github.io/#packet-core.packet describes the `core.packet` API. It doesn't mention the actual packet structure which is why I am uneasy about using the fields directly instead of going through the API functions.